app/elm/Images/GalleryView.elm

```diff
@@ -0,0 +1,74 @@
+module Images.GalleryView exposing (galleryView)
+
+import Html exposing (Html, div, a, img, text)
+import Html.Attributes exposing (class, href, src, alt, title)
+import Routing exposing (imagePath, personPath, personImagePath)
+import Images.Models exposing (..)
+import Images.Utils exposing (..)
+
+
+galleryView : Maybe PersonId -> ( List Image, List Person ) -> Html msg
+galleryView maybePersonId ( allImages, allPeople ) =
+    case maybePersonId of
+        Just personId ->
+            case (getPerson personId allPeople) of
+                Just person ->
+                    (getImagesOfPerson personId allImages)
+                        |> List.map (cardView maybePersonId)
+                        |> buildGalleryView
+
+                Nothing ->
+                    notFoundView
+
+        Nothing ->
+            allImages
+                |> List.map (cardView Maybe.Nothing)
+                |> buildGalleryView
```

Comment:
I suspect there's a little bit of redundancy here that could be removed, but I'm too sleepy to figure it out right now. 😪